lesson3/Lesson_3/app/src/main/java/mobpro/lesson_3/MainActivityFragment.java

```diff
@@ -180,16 +80,28 @@ public void onClick(View v) {
 
         return view;
     }
-}
 
-//    Button switchToSettings = (Button) view.findViewById(R.id.switchButton); //instantiate prompt button w/ button id from fragment_main.xml
-//switchToSettings.setOnClickListener(new View.OnClickListener() {
-//@Override
-//public void onClick(View v) {
-//        Fragment fragment = new Fragment();
-//        android.support.v4.app.FragmentManager fragmentManager = getFragmentManager();
-//        fragmentManager.beginTransaction()
-//        .replace(R.id.settings_fragment, fragment)
-//        .commit();
-//        }
-//        });
\ No newline at end of file
+    private void showAlertDialog(final TextView textview) {
+        AlertDialog.Builder popup_maker = new AlertDialog.Builder(getContext());
+        popup_maker //configure popup
+                .setTitle("Do you want to change this list item?")
+                .setMessage("Click SET to save your change.")
+                .setCancelable(true);
+        final EditText field = new EditText(getContext());
+        popup_maker.setView(field);
+        popup_maker.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.dismiss();
+            }
+        });
+        popup_maker.setPositiveButton("Set", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                textview.setText(field.getText().toString());
+            }
+        });
+        AlertDialog popup = popup_maker.create();
+        popup.show();
+    }
+}
```

Comment:
Instead of copy-pasting each AlertDialog code, abstract it to one function!  Here, the only thing that changed between each TextView's onClickListener is the TextView to call `.setText()` on, so just pass in the specific TextView in the abstracted function, as I did here.